ci: auto generate integrations mapping table

0xmayro opened the pull request:
I accidentally messed up the [old PR](https://github.com/catppuccin/nvim/pull/881) trying to fix a different one.

this should be the same.

0xmayro committed:
feat: generation script for plugin_integration table

0xmayro committed:
refactor: use generated table from `generate_integration_mappings.lua` script

0xmayro committed:
fix: check before requiring integration_mappings.lua

0xmayro committed:
refactor: use fields instead of comments

I wrapped almost the entire feline integration feline in pcall to enable
the usage of this method.

0xmayro committed:
fix: extract plugin name from the field

0xmayro committed:
fix: remove unused variable

0xmayro committed:
ci: integration table generation

orginally by @vollowx. I messed up with rebasing and need to recommit to
fix it

0xmayro committed:
fix: adhere to stylua formatting for generated file

0xmayro commented:
@vollowx @sgoudham , again I'm so sorry for all the mess.

vollowx reviewed (approved):
LGTM 👍

comfysage commented on scripts/generate_integration_mappings_table.lua:
```diff
@@ -0,0 +1,51 @@
+		if ok then
+			---@type string
+			local plugin_name = mod.url
+			plugin_name = plugin_name:match ".*/(.-)/*$" or plugin_name -- extract the repo from the url. else use the url.
```
can this pattern be replaced with `([^/]+)$`?

comfysage commented on scripts/generate_integration_mappings_table.lua:
```diff
@@ -0,0 +1,51 @@
+		local ok, mod = pcall(require, "catppuccin.groups.integrations." .. filename)
+
+		if ok then
+			---@type string
+			local plugin_name = mod.url
+			plugin_name = plugin_name:match ".*/(.-)/*$" or plugin_name -- extract the repo from the url. else use the url.
+			mappings[plugin_name] = filename
```
```suggestion
			do
				---@type string
				local plugin_name = mod.url
				if not plugin_name then
					return
				end
				plugin_name = plugin_name:match ".*/(.-)/*$" or plugin_name -- extract the repo from the url. else use the url.
				mappings[plugin_name] = filename
			end
```

0xmayro replied:
this is for making sure mod.url is correctly set, right?
instead of `return` should it print an error with the file name?

comfysage reviewed (changes requested):
looks great :) thnx for all the hard work

just some small changes id like to see

0xmayro committed:
fix: changed lua pattern

comfysage committed:
Apply suggestion from @comfysage

comfysage commented:
summary of required changes (as described in collaboration with @sgoudham ):
- move logic out to utility function (e.g., getting the name from url) and write tests.
- compare `M.url` with the name of the repo according to the plugin-spec's urls from `vim.pack`, `lazy.nvim`, `pckr`.
- merge #910 into this pr.

if you need help with any of these id love to help write some of the logic/tests

0xmayro commented:
about this, should the utility function be defined in a new file? or can the script return it?
i.e. 
```lua
-- script.lua
local M = {}
function M.get_name(url)
	return url:match("([^/]+)$")
end
-- rest of script as defined in PR...

return M
```

0xmayro committed:
refactor: moved url parsing to it's own function and added tests

0xmayro committed:
fix(tests)

comfysage committed:
merge additional plugin managers (#6)

Co-authored-by: Mayro <may.ronen9@gmail.com>

comfysage committed:
refactor: add `detect_plugins` fn

0xmayro committed:
fix: pass stylua

comfysage committed:
refactor: move util fn to lib

- rewrite tests
- move `parse_url`

comfysage committed:
fix: remove fake plugin urls

- remove `M.url` from `markdown` and `semantic_tokens`
- use `Iter:fold()` for `scripts/generate_integration_mappings_table.lua`
- allow for `nil` urls in `scripts/generate_integration_mappings_table.lua`

comfysage committed:
style: fmt

comfysage committed:
fix(workflows): add ref to checkout

sgoudham reviewed (approved):
Trusting that it's good to merge from @comfysage
